content/en/docs/tasks/administer-cluster/nodelocaldns.md

```diff
@@ -47,18 +47,37 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
 {{< figure src="/images/docs/nodelocaldns.jpg" alt="NodeLocal DNSCache flow" title="Nodelocal DNSCache flow" caption="This image shows how NodeLocal DNSCache handles DNS queries." >}}
 
 ## Configuration
-
-This feature can be enabled using the command:
-
-`KUBE_ENABLE_NODELOCAL_DNS=true kubetest --up`
-
-This works for e2e clusters created on GCE. On all other environments, the following steps will setup NodeLocal DNSCache:
-
-* A yaml similar to [this](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml) can be applied using `kubectl create -f` command.
-* No need to modify the --cluster-dns flag since NodeLocal DNSCache listens on both the kube-dns service IP as well as a link-local IP (169.254.20.10 by default)
+NOTE: The local listen IP address for NodeLocal DNSCache can be any IP in the 169.254.20.0/16 space or any other IP address that can be guaranteed to not collide with any existing IP. We will use 169.254.20.10 as the default in the example below.
+
+This feature can be enabled using the following steps:
+* Prepare a yaml similar to [this](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml), save as nodelocaldns.yaml.
+* Substitute the variables in the yaml with the right values:
+   kubedns=`kubectl get svc kube-dns -n kube-system -o jsonpath={.spec.clusterIP}`
+   domain="cluster.local"
+   localdns="169.254.20.10"
+
+   * If kube-proxy is running in iptables mode: 
+   
+     ```
+     sed -i "s/__PILLAR__LOCAL__DNS__/$localdns/g; s/__PILLAR__DNS__DOMAIN__/$domain/g; s/__PILLAR__DNS__SERVER__/$kubedns/g" nodelocaldns.yaml
+     ```
+     
+    `__PILLAR__CLUSTER__DNS__` and `__PILLAR__UPSTREAM__SERVERS__` will be populated by the node-local-dns pods.  
+    In this mode, node-local-dns pods listen on both the kube-dns service IP as well as the link-local address(169.254.20.10), so pods can lookup DNS records using either IP address.
+
+  * If kube-proxy is running in ipvs mode: 
+  
+    ```
+     sed -i "s/__PILLAR__LOCAL__DNS__/$localdns/g; s/__PILLAR__DNS__DOMAIN__/$domain/g; s/__PILLAR__DNS__SERVER__//g; s/__PILLAR__CLUSTER__DNS__/$kubedns/g" nodelocaldns.yaml
+    ```
+    In this mode, node-local-dns pods listen only on the link-local address(169.254.20.10). The node-local-dns interface cannot bind the kube-dns cluster ip since the interface used for IPVS loadbalancing already uses this address.  
```

Comment:
```suggestion
    In this mode, node-local-dns pods listen only on the link-local address (169.254.20.10). The node-local-dns interface cannot bind the kube-dns cluster IP since the interface used for IPVS loadbalancing already uses this address.
```

Reply:
Done